src/java/org/apache/cassandra/transport/Dispatcher.java

```diff
@@ -69,7 +67,10 @@ public Dispatcher(boolean useLegacyFlusher)
 
     public void dispatch(Channel channel, Message.Request request, FlushItemConverter forFlusher, Overload backpressure)
     {
-        Stage.NATIVE_TRANSPORT_REQUESTS.submit(() -> processRequest(channel, request, forFlusher, backpressure));
+        if (request.type == Message.Type.QUERY)
```

Comment:
@jakubzytka please double check if this is the correct place for doing this. There is also this https://github.com/riptano/cndb/blob/aa4ceea499e67b6befabef8342544f9a49e3eec6/cndb-stargate/cndb-persistence/src/main/java/io/stargate/db/dse/impl/CndbPersistence.java#L576

Reply:
Changes to Stargate are here: https://github.com/riptano/cndb/pull/10905

Reply:
Please forgive my ignorance, but how does `getRequestStage` distinguish reads from other requests?

E.g. if `request.type == Message.Type.QUERY  <=> this is a read request`, then what is the `Message.Type` of a write request?
Of a range read request? 
What if `request.type == EXECUTE`? Are prepared statements handled correctly?

Reply:
Agreed. Pretty sure simple string INSERT and SELECT (and all other non-prepared queries) are of type QUERY, and for most workloads EXECUTE will be the main type also with both INSERT and SELECT in it.

Reply:
Yes you are obviously both right, I should have not done this on a Friday evening, so pardon my lapse of reason. Pushed a proper version which is (unfortunately?) way more involving than I was hoping for.